dbt/adapters/duckdb/plugins/excel.py

```diff
@@ -29,3 +40,45 @@ def load(self, source_config: SourceConfig):
             source_location = pathlib.Path(ext_location.strip("'"))
         sheet_name = source_config.get("sheet_name", 0)
         return pd.read_excel(source_location, sheet_name=sheet_name)
+
+
+    def store(self, target_config: TargetConfig):
+        plugin_output_config = self._config["output"]
+
+        # Create the writer on the first instance of the call to store.
+        # Instead if we instantiated the writer in the constructor
+        # with mode = 'w', this would result in an existing file getting
+        # overwritten. This can happen if dbt test is executed for example.
+        if not hasattr(self, '_excel_writer'):
```

Comment:
Q: does this bit need a lock? I'm assuming it's possible for multiple threads to hit it at the same time.